lxd/network/driver_bridge.go

```diff
@@ -3050,7 +3050,17 @@ func (n *bridge) getExternalSubnetInUse() ([]externalSubnetUsage, error) {
 func (n *bridge) ForwardCreate(forward api.NetworkForwardsPost, clientType request.ClientType) (net.IP, error) {
 	memberSpecific := true // bridge supports per-member forwards.
 
-	err := n.state.DB.Cluster.Transaction(context.TODO(), func(ctx context.Context, tx *db.ClusterTx) error {
+	// Convert listen address to subnet so we can check its valid and can be used.
+	listenAddressNet, err := ParseIPToNet(forward.ListenAddress)
+	if err != nil {
+		return nil, fmt.Errorf("Failed parsing address forward listen address %q: %w", forward.ListenAddress, err)
+	}
+
+	if listenAddressNet.IP.IsUnspecified() {
+		return nil, api.StatusErrorf(http.StatusNotImplemented, "Automatic listen address allocation not supported for drivers of type %q", n.netType)
+	}
+
+	err = n.state.DB.Cluster.Transaction(context.TODO(), func(ctx context.Context, tx *db.ClusterTx) error {
 		// Check if there is an existing forward using the same listen address.
 		_, _, err := tx.GetNetworkForward(ctx, n.ID(), memberSpecific, forward.ListenAddress)
```

Comment:
Can we restructure this part to use the `externalSubnetsInUse` list (like you did for the `ovn` driver), just without adding support for automatic allocation, that way the structure of the code and the errors remains the same for each driver?